src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,48 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
 
+    public MateStack() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == elements.length) {
+            increaseCapacity();
+        }
+        elements[++size] = value;
```

Comment:
```suggestion
        elements[size++] = value;
```
when size is 0 you should put an element on the 0 index not 1